pkg/packing/packable.go

```diff
@@ -100,10 +101,29 @@ func (p *Packable) Pack(pods []*v1.Pod) *Result {
 			return result
 		}
 		result.unpacked = append(result.unpacked, pod)
+
+		if p.isFull(pods[len(pods)-1]) {
+			result.unpacked = append(result.unpacked, pods[i+1:]...)
+			return result
+		}
 	}
 	return result
 }
 
+// isFull checks if the reserved resources + the minimum resourced pod overflows the total resources available.
+func (p *Packable) isFull(minPod *v1.Pod) bool {
```

Comment:
Any reason you can't use 
```
func (p *Packable) reservePod(pod *v1.Pod) bool {
	requests := resources.RequestsForPods(pod)
	requests[v1.ResourcePods] = *resource.NewQuantity(1, resource.BinarySI)
	return p.reserve(requests)
}
func (p *Packable) reserve(requests v1.ResourceList) bool {
	candidate := resources.Merge(p.reserved, requests)
	// If any candidate resource exceeds total, fail to reserve
	for resourceName, quantity := range candidate {
		if quantity.Cmp(p.total[resourceName]) > 0 {
			return false
		}
	}
	p.reserved = candidate
	return true
}
```

As far as I understand it, this function does exactly what you need.

Reply:
Yeah, you're probably right that this can be reduced. Let me try to merge these. I think the dry-run would be the isFull and the reserve would still exist but do less and just call isFull and the reserve.

Reply:
actually, after digging into trying to merge these two or use each other, I think it's actually making the interface of reserve significantly worse off. The reserve function is doing something slightly different and returning a different result. 

The reserve function will take a resource list and reserve on the node if it doesn't overflow the node's resource. The return value signifies that the reservation succeeded (did not overflow) or failed (overflowed).

The isFull function is only concerned about the node totals and seeing they are all the way full, but not overflowed OR would overflow with the smallest pod that needs to be reserved in the list. 

I think merging these functions makes it less readable, adds complexity to reserve, and the return value gets muddied since we'd need two bools returns, one indicating if the reservation succeeded and one noting if the node is full.